app/models/track.rb

```diff
@@ -41,6 +41,11 @@ def git
     Git::Track.new(synced_to_git_sha, repo_url: repo_url)
   end
 
+  memoize
+  def repo_name
+    slug
+  end
```

Comment:
You should get this from the `repo_url`, not the slug. Sometimes they get mismatched, which could be bad for us here.